+ source /root/mugen/libs/locallibs/common_lib.sh
++ python3 --version
++ '[' 0 -eq 0 ']'
++ source /root/mugen/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Thu Aug  3 18:38:55 2023 - INFO  - Start environmental preparation.
+ DNF_INSTALL 'hdf5 hdf5-devel'
+ pkgs='hdf5 hdf5-devel'
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen/libs/locallibs/rpm_manage.py install --pkgs 'hdf5 hdf5-devel' --node 1 --tempfile ''
+ tmpfile2=/tmp/tmp03u49ryh
+ '[' -z '' ']'
+ tmpfile=/tmp/tmp03u49ryh
+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Thu Aug  3 18:39:37 2023 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start testing...'
+ message='Start testing...'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start testing...'
Thu Aug  3 18:39:38 2023 - INFO  - Start testing...
+ cat
+ ls hdf5_example.c
hdf5_example.c
+ CHECK_RESULT 0 0 0 'Create hdf5_example.c file fail'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Create hdf5_example.c file fail'
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ gcc hdf5_example.c -o hdf5_example -lhdf5
In file included from /usr/include/bits/posix1_lim.h:161,
                 from /usr/include/limits.h:195,
                 from /usr/lib/gcc/riscv64-linux-gnu/10.3.1/include/limits.h:195,
                 from /usr/lib/gcc/riscv64-linux-gnu/10.3.1/include/syslimits.h:7,
                 from /usr/lib/gcc/riscv64-linux-gnu/10.3.1/include/limits.h:34,
                 from /usr/include/H5public.h:41,
                 from /usr/include/hdf5.h:22,
                 from hdf5_example.c:3:
/usr/include/bits/local_lim.h:38:10: fatal error: linux/limits.h: No such file or directory
   38 | #include <linux/limits.h>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
+ CHECK_RESULT 1 0 0 'Compile fail'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Compile fail'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Compile fail'
+ LOG_ERROR 'Compile fail'
+ message='Compile fail'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Compile fail'
Thu Aug  3 18:39:39 2023 - ERROR - Compile fail
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_hdf5_01.sh line 62'
+ message='oe_test_hdf5_01.sh line 62'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_hdf5_01.sh line 62'
Thu Aug  3 18:39:40 2023 - ERROR - oe_test_hdf5_01.sh line 62
+ '[' 0 -eq 1 ']'
+ return 0
+ ./hdf5_example
oe_test_hdf5_01.sh: line 63: ./hdf5_example: No such file or directory
+ CHECK_RESULT 127 0 0 'Execute fail'
+ actual_result=127
+ expect_result=0
+ mode=0
+ error_log='Execute fail'
+ exit_mode=0
+ '[' -z 127 ']'
+ '[' 0 -eq 0 ']'
+ test 127x '!=' 0x
+ test -n 'Execute fail'
+ LOG_ERROR 'Execute fail'
+ message='Execute fail'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Execute fail'
Thu Aug  3 18:39:40 2023 - ERROR - Execute fail
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_hdf5_01.sh line 64'
+ message='oe_test_hdf5_01.sh line 64'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_hdf5_01.sh line 64'
Thu Aug  3 18:39:41 2023 - ERROR - oe_test_hdf5_01.sh line 64
+ '[' 0 -eq 1 ']'
+ return 0
+ ls example.h5
ls: cannot access 'example.h5': No such file or directory
+ CHECK_RESULT 2 0 0 'Create example.h5 file fail'
+ actual_result=2
+ expect_result=0
+ mode=0
+ error_log='Create example.h5 file fail'
+ exit_mode=0
+ '[' -z 2 ']'
+ '[' 0 -eq 0 ']'
+ test 2x '!=' 0x
+ test -n 'Create example.h5 file fail'
+ LOG_ERROR 'Create example.h5 file fail'
+ message='Create example.h5 file fail'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Create example.h5 file fail'
Thu Aug  3 18:39:41 2023 - ERROR - Create example.h5 file fail
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_hdf5_01.sh line 66'
+ message='oe_test_hdf5_01.sh line 66'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_hdf5_01.sh line 66'
Thu Aug  3 18:39:42 2023 - ERROR - oe_test_hdf5_01.sh line 66
+ '[' 0 -eq 1 ']'
+ return 0
+ LOG_INFO 'Finish test!'
+ message='Finish test!'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Finish test!'
Thu Aug  3 18:39:42 2023 - INFO  - Finish test!
+ CASE_RESULT 0
+ case_re=0
+ test -z 3
+ test 3 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Thu Aug  3 18:39:43 2023 - ERROR - failed to execute the case.
+ exit 3
+ post_test
+ LOG_INFO 'start environment cleanup.'
+ message='start environment cleanup.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'start environment cleanup.'
Thu Aug  3 18:39:43 2023 - INFO  - start environment cleanup.
+ rm -fr hdf5_example.c hdf5_example example.h5
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z /tmp/tmp03u49ryh ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile /tmp/tmp03u49ryh
+ '[' 0 -ne 0 ']'
+ LOG_INFO 'Finish environment cleanup!'
+ message='Finish environment cleanup!'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Finish environment cleanup!'
Thu Aug  3 18:39:52 2023 - INFO  - Finish environment cleanup!
